### DISCUSSION TITLE
SSh Key error private repository on github

### DISCUSSION THREAD

**Discussion by carlosecv** [Tecnativa/doodba#431] [Doodba - The template] [answered]
1.- I created my copier template project on my desktop ~/project/doodba-virtual/
2.- Created my ssh keys for my private repository on github
     ssh-keygen -t ed25519 -C "myuser@mydomain.com" -f ~/project/doodba-virtual/odoo/custom/ssh/id_rsa
3.- I created a private repository on github 
4.- Then add the public key from my user on github 
5.- Then i created my config file on ssh folder, 

Whenever i try to get the repositories on my copier template doodba project i get error for ssh keys:

My file on ~/project/doodba-virtual/odoo/custom/ssh/config:

Host github.com
    HostName github.com
    User myuser
    IdentityFile /opt/odoo/custom/ssh/id_rsa

When i do invoke git-aggregate:

Warning: Permanently added the RSA host key for IP address '140.82.113.X' to the list of known hosts.
no such identity: /opt/odoo/custom/ssh/id_rsa: Permission denied
myuser@github.com: Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

If i do the same with private repository on my own gitlab server tehere is no error but on github.com or gitlab.com get the same error, is something im missing?

Thank you

**Comment by andhit-r** [comment]
Try run invoke img-build before invoke git-aggregate

**Comment by carlosecv** [comment, marked as answer]
Thank you andhit-r I found the answer:

on my config file i had:

IdentityFile **/opt/odoo/custom/ssh**/id_rsa

I changed by 

IdentityFile ~/.ssh/id_rsa

Then I run invoke img-build then invoke git-aggregate and  and it worked
